### DISCUSSION TITLE
How Release note are polulated

### DISCUSSION THREAD

**Discussion by mfvitale** [renovatebot/renovate#15265] [Request Help]
Hi all, 

I have a simple question: How the section Release note is populated? I have some internal repositories with release notes but I don't see it when PR is opened on dependent projects. I don't see either "(source)" or Release note section. What is missing? 

This is an example of what I would like to have for internal dependencies. Do I need to configure something on pom.xml?

![image](https://user-images.githubusercontent.com/2572323/164714355-94b8f6bc-41c7-4eb9-8948-77c3fa81ad03.png)

Thanks,
Mario.

**Comment by viceice** [comment]
it depends on renovate manager / datasource usage, so it looks like you are using maven / gradle?

so we use our `maven` datasource there:

https://github.com/renovatebot/renovate/blob/ee582fe0e7c06e8c2a1d056cb280e8558dab30d8/lib/modules/datasource/maven/util.ts#L224-L238

renovate requires `sourceUrl` to fetch release notes (only github and gitlab are supported for release notes)

**Reply by mfvitale** [reply]
Yes, I am using maven. I have set the URL so now I am able to see the correct link to the source but still no release notes. 

We are using GitHub enterprise. Can this be the problem?

**Reply by viceice** [reply]
Here it detects if it's possible to fetch releasenotes: 


https://github.com/renovatebot/renovate/blob/fb9303c19081e341c123581c00b3e72701c1c8c9/lib/modules/platform/util.ts#L14-L38


url is the extracted `sourceUrl`

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/fb9303c19081e341c123581c00b3e72701c1c8c9/lib/workers/repository/update/pr/changelog/index.ts#L31-L47

**Reply by mfvitale** [reply]
Where 'sourceUrl' comes from?

In the spec **https://github.com/renovatebot/renovate/blob/fb9303c19081e341c123581c00b3e72701c1c8c9/lib/modules/platform/util.spec.ts** 
I don't see something like 'https://github.enterprise.com/myorg/myrepo. Should it be a valid GitHub platform?

**Reply by viceice** [reply]
`sourceUrl` comes from datasource, see first comment. Please check the debug logs, are there any messages about chagelogs?
